website/docs/generics.md

```diff
@@ -0,0 +1,1102 @@
+
+  This is bad because the generic abstraction being built will not necessarily
+  provide the guarantees it should. This can give users of the generic
+  abstraction false confidence in the type system.
+
+  To mitigate this, write example code that should not type check and double
+  check that it doesn't. Get creative with these tests. Consider writing tests
+  that make uncommon use of subtyping, inheritance, mutation, etc.
+
+  (There is nothing built into Sorbet for writing such tests. The easiest
+  approach is to manually build small examples using the new API that don't type
+  check, but don't check the resulting files in (or check them in, but mark them
+  `# typed: ignore`, and bump the sigil up temporarily while making changes).
+  The diligent may way to automate this by running Sorbet a second time on a
+  codebase that includes extra files meant to not type check, asserting that
+  Sorbet indeed reports errors.)
+
+  It can also be helpful to use `T.reveal_type` and/or `T.assert_type!` to
+  inspect the types of intermediate values to see if `T.untyped` has silently
+  snuck in somewhere.
+
+- Many things **don't type check when they should**.
+
+  This is bad because it causes confusion and frustration for people attempting
+  to use the generic abstraction, not for the person who implemented the
+  abstraction. This is especially painful for those who are new to Sorbet (or
+  even Ruby), as well as those those who are not intimately familiar with the
+  limitations of Sorbet's generics.
+
+  To mitigate this, "test drive" the abstraction being built. Don't assume that
+  if the implementation type checks that it will work for downstream consumers.
+  Get creative and write code as a user would use the abstraction. Encountering
+  these errors is not only frustrating for you, but also frustrating for others,
+  and incurs a real risk of making people's first experience with Sorbet unduly
+  negative.
+
+Both of these outcomes are bad, and both are unfortunately quite common. When
+building new generics abstractions, please keep these in mind. We apologize for
+the inconvenience this causes.
+
+As with all bugs in Sorbet, when you encounter them [please report
+them][issues]. See the list of known bugs here:
+
+→ [Generics milestone]
+
+## Basic syntax
+
+The basic syntax for class generics in Sorbet looks like this:
+
+```ruby
+# typed: strict
+
+class Box
+  extend T::Sig
+  extend T::Generic # Provides `type_member` helper
+
+  Elem = type_member # Makes the `Box` class generic
+
+  # References the class-level generic `Elem`
+  sig {params(val: Elem).void}
+  def initialize(val:); @val = val; end
+  sig {returns(Elem)}
+  def val; @val; end
+  sig {params(val: Elem).returns(Elem)}
+  def val=(val); @val = val; end
+end
+
+int_box = Box[Integer].new(val: 0)
+T.reveal_type(int_box) # `Box[Integer]`
+
+T.reveal_type(int_box.val) # `Integer`
+
+int_box.val += 1
+
+string_box = int_box.map {|val| val.to_s}
```

Comment:
Running the given example in `sorbet.run`:

```
editor.rb:25: Method map does not exist on Box[Integer] https://srb.help/7003
    25 |string_box = int_box.map {|val| val.to_s}
                             ^^^
  Got Box[Integer] originating from:
    editor.rb:18:
    18 |int_box = Box[Integer].new(val: 0)
                  ^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
whoops, i made this edit in response to some other comments, forgot to update the example. thanks!